#!/usr/bin/env python
PACKAGE = "robotis_op_simulation_walking"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# see http://support.robotis.com/en/product/darwin-op/development/tools/walking_tuner.htm for detailed documentation
# Walking initial pose
gen.add("X_OFFSET",     double_t,  0, "X offset(mm)",	-10.0, -50, 50)
gen.add("Y_OFFSET",     double_t,  0, "Y offset(mm)",	5.0, -50, 50)
gen.add("Z_OFFSET",     double_t,  0, "Z offset(mm)",	20.0, -50, 50)
gen.add("R_OFFSET",   	double_t,  0, "Roll offset(degree)",	0, -30, 30)
gen.add("P_OFFSET",     double_t,  0, "Pitch offset(degree)",     0, -30, 30)
gen.add("A_OFFSET",   	double_t,  0, "Yaw offset(degree)",	0, -30, 30)
#Walking control
gen.add("PERIOD_TIME",    double_t,  0, "Period time(msec)",     		1000, 100, 10000)
gen.add("DSP_RATIO",    double_t,  0, "Double stance period ratio",     		0.1, 0, 2)
gen.add("STEP_FB_RATIO",    double_t,  0, "Step forward/back ratio",     		0.28, 0, 5)
gen.add("Z_MOVE_AMPLITUDE",    double_t,  0, "Z move amplitute(mm)",     		60, 0, 100)
gen.add("Y_SWAP_AMPLITUDE",    double_t,  0, "Y swing amplitute(mm)",     		20, 0, 100)
gen.add("Z_SWAP_AMPLITUDE",    double_t,  0, "Z swing amplitute(mm)",     		5, 0, 100)
gen.add("PELVIS_OFFSET",    double_t,  0, "Pelvis offset(degree)",     3, 0, 100)
gen.add("ARM_SWING_GAIN",    double_t,  0, "Arm swing gain",     1.5, 0, 100)
gen.add("BALANCE_KNEE_GAIN",    double_t,  0, "Balance knee gain",     0.3, 0, 100)
gen.add("BALANCE_ANKLE_PITCH_GAIN",    double_t,  0, "Balance ankle pitch gain",     		0.9, 0, 100)
gen.add("BALANCE_HIP_ROLL_GAIN",    double_t,  0, "Balance hip roll gain",     		0.5, 0, 100)
gen.add("BALANCE_ANKLE_ROLL_GAIN",    double_t,  0, "Balance ankle roll gain",     		1.0, 0, 100)
gen.add("HIP_PITCH_OFFSET",       double_t,  0, "Hip pitch offset(degree)",     		12.0, 0, 100)

exit(gen.generate(PACKAGE, "robotis_op_simulation_walking", "robotis_op_walking"))
